### PR TITLE
Validates the adding of Topics

### PR DESCRIPTION
**Pull request by tylerpendleton**
Doesn't allow blank topics to be created
screen shot 2015-09-11 at 2 55 20 am

### CODE REVIEW

File: app/controllers/talks_controller.rb [hunk 1]
```diff
@@ -12,8 +12,12 @@ def new
   end
 
   def create
-    current_user.talks.create(talk_params)
-    redirect_to root_path
+    @talk = current_user.talks.create(talk_params)
+    if @talk.valid?
+      redirect_to root_path
+    else
+      render :new, :status => :unprocessable_entity
```
Comment (colinrubbert) [inline app/controllers/talks_controller.rb]: Style for Ruby 2.0+ 
`render :new, status: :unprocessable_entity`

File: app/models/talk.rb [hunk 2]
```diff
@@ -5,4 +5,8 @@ class Talk < ActiveRecord::Base
   scope :unscheduled, -> { where(speak_date: nil).order(:topic) }
   scope :scheduled, -> { where('speak_date >= ?', Date.today).order(:speak_date).order(:topic) }
   scope :previous, -> { where('speak_date < ?', Date.today).order(:topic) }
+
+  # Add Validation
+  validates :topic, :presence => true, length: { minimum: 3 }
```
Comment (colinrubbert) [inline app/models/talk.rb]: Styling for Ruby
`validates :topic, presence: true, length: { minimum: 3 }`

File: app/models/talk.rb [hunk 3]
```diff
@@ -5,4 +5,8 @@ class Talk < ActiveRecord::Base
...
+  validates :description, :presence => true
```
Comment (colinrubbert) [inline app/models/talk.rb]: `validates :topic, presence: true, length: { minimum: 3 }`

File: test/controllers/talks_controller_test.rb [hunk 4]
```diff
@@ -1,7 +1,17 @@
...
-  # test "the truth" do
-  #   assert true
-  # end
+
+  test "do not add blank topics" do
+    user = FactoryGirl.create(:user)
+    sign_in user
+
+    assert_no_difference "Talk.count" do
+      post :create, :talk => {
```
Comment (colinrubbert) [inline test/controllers/talks_controller_test.rb]: Styling for Ruby 2.0+ 
fix the `=>` if you can

**Comment by colinrubbert** [comment]
@tpendleton1693 looks good, just go in and fix the `=>` to Ruby 2.0+ style guides and I'll merge it into master. Again I'm just nitpicking on the styling, the code is solid.

**Comment by kenmazaika** [comment]
@colinrubbert is such a stickler for the hash rocket syntax, haha. :)

**Comment by colinrubbert** [comment]
@kenmazaika I'm sneakily getting people into best practices for other OSS projects. 

The lightning talk and community apps are the pre-course work for the wider OSS community :smile:

**Comment by tylerpendleton** [comment]
I had no idea that was a best practice! Is there a guide I can look at
somewhere?

**Comment by colinrubbert** [comment]
@tpendleton1693 here is GitHub's [styleguide](https://github.com/styleguide/ruby) which contains a bunch of bad/good examples for Ruby styling, which actually shows quite a bit of use on still using `=>`. 

I don't know that it's necessarily "best practice", however, when I look through other OSS projects, especially well developed ones, you rarely see the `=>` anymore (which to me means it's "best practice"). Most of it will be based off of your team's styling guide. Since I see more people using the `=>` less and less in the OSS community I figured that was a best approach. 

it's all a bit confusing and no one has a real clear "YES DO IT THIS WAY!" answer but you definitely shouldn't use both in your code. Either do it with all `=>` or none, I started the app with the decision not to and to protect everyone's sanity going forward we'll keep it that way unless someone can give me a good reason not. :smile:

**Comment by kenmazaika** [comment]
Either way works, but the rule of thumb is to pick one and consistently use that way.

**Comment by kenmazaika** [comment]
Whoops I accidentally closed this.  Didn't mean to.

**Comment by tylerpendleton** [comment]
I updated the branch!  No more =>

File: config/database.example.yml [hunk 5]
```diff
@@ -1,21 +0,0 @@
-default: &default
```
Comment (kenmazaika) [inline config/database.example.yml:1]: Looks like this file was removed on accident.  We should keep this file in.

**Comment by kenmazaika** [comment]
Made one note.  Besides that :+1: from me.

**Comment by tylerpendleton** [comment]
Ahh! i'll put it back.  Thanks Ken!

**Comment by colinrubbert** [comment]
@tpendleton1693 we definitely want to keep that `database.example.yml` file. That's part of the setup process. 
I'll test and verify everything is working but from just looking at the code it looks good. It will likely be merged today.

**Comment by tylerpendleton** [comment]
@colinrubbert Awesome.  It feels good to contribute!

**Comment by colinrubbert** [comment]
@tpendleton1693 helps me out a ton too so I am very grateful! You're doing good work and making great steps towards being able to work on other OSS projects.

**Comment by colinrubbert** [comment]
@tpendleton1693 Great job on the PR!

**Comment by tylerpendleton** [comment]
Thanks!

**Pull request merged**